lib/syntax_tree/node.rb

```diff
@@ -848,6 +852,22 @@ def format(q)
     def ===(other)
       other.is_a?(Args) && ArrayMatch.call(parts, other.parts)
     end
+
+    def arity
+      accepts_infinite_arguments? ? Float::INFINITY : parts.length
```

Comment:
Kept infinity here to represent an splats and argument forwarding, so that we can do `def_node.arity.cover?(call_node.arity)`.